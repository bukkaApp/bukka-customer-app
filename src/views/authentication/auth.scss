.auth-page {
    @media (max-width: 576px) {
       background-color: white!important;
    }
    @media (max-width: 767px) {
       min-height: calc(#{"100vh - 65px"});
    }
}

.auth-wrapper {
    @media (max-width: 576px) {
        box-shadow: none!important;
     }
}
