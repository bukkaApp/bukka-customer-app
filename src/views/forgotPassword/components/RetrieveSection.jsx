import React, { useState, useEffect } from 'react';

import { useHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Navbar from 'Components/navbar';
import Container from 'Components/container';
import AuthResetForm from './AuthResetForm';
import TokenErrorFeedback from './TokenErrorFeedback';

import ModalRoot from '../../modal-root/Index';
import changePassword from '../actionCreators/changePassword';
import validateToken from '../actionCreators/validateToken';
import { validateAField, validateAllFields } from '../helper/validateFields';

import signUpDomStructure from '../signUpDomStructure.json';
import useQuery from '../../../context/useQuery';

const RetrieveSection = ({
  isValidUser,
  requestSent,
  checkIsValidUser,
  requested,
  errorMessage,
  changeAuthPassword,
}) => {
  const query = useQuery();
  const { push } = useHistory();
  const [validationErrors, setValidationErrors] = useState({
    password: '',
    confirmPassword: ''
  });

  const [inputData, setInputData] = useState({
    password: '',
    confirmPassword: ''
  });

  const validateOnClick = (newValidationErrors) => {
    setValidationErrors({
      ...validationErrors,
      ...newValidationErrors
    });
  };

  const handleChange = ({ target: { name, value } }) => {
    const newFieldData = { [name]: value };
    const validation = validateAField(newFieldData, name);
    setInputData({
      ...inputData,
      ...newFieldData
    });
    setValidationErrors({
      ...validationErrors,
      [name]: validation.message
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const validation = validateAllFields(inputData);

    const { errors, passes } = validation;
    validateOnClick(errors);
    if (passes) {
      return changeAuthPassword('/user/reset', inputData, query.get('token'));
    }
  };

  const handleToken = (token) => {
    if (!token) {
      return push('/');
    }
  };

  const handleUserValidity = async (token) => {
    if (!isValidUser) {
      handleToken(token);
      await checkIsValidUser('/user/token/validate', token);
    }
  };

  const successRedirection = () => {
    if (requested) {
      return push('/login');
    }
  };

  useEffect(() => {
    handleUserValidity(query.get('token'));
    successRedirection();
  });

  return (
    <>
      <ModalRoot push={push} />
      <Navbar push={push} />
      <div className="bg-color py-8">
        <Container classNames="relative modal-open">
          <div
            className="d-flex flex-column flex-xl-row
              flex-lg-row flex-md-column justify-content-center"
          >
            <div
              className="col-xl-6 col-lg-6 px-0
                px-md-0 px-lg-3 col-md-12 col-12"
            >
              {requestSent &&
              <div className="card-shadow card mb-3 border">
                {isValidUser ? (
                  <AuthResetForm
                    push={push}
                    title="Reset Password"
                    errorMessage={errorMessage}
                    instruction="Type your new password"
                    handleChange={handleChange}
                    validationErrors={validationErrors}
                    handleSubmit={handleSubmit}
                    domStructure={signUpDomStructure}
                    isFormCompleted
                  />
                ) : (
                  <TokenErrorFeedback push={push} />
                )}
              </div>
              }
            </div>
          </div>
        </Container>
      </div>
    </>
  );
};

const mapStateToProps = ({
  changePasswordReducer: { requested, errorMessage },
  validateTokenReducer: { isValidUser, requested: requestSent }
}) => ({
  isValidUser,
  requestSent,
  requested,
  errorMessage
});

export default connect(mapStateToProps, {
  changeAuthPassword: changePassword,
  checkIsValidUser: validateToken
})(RetrieveSection);

RetrieveSection.defaultProps = {
  changeAuthPassword: () => {},
  checkIsValidUser: () => {},
  requested: false,
  requestSent: false,
  isValidUser: false,
  errorMessage: ''
};

RetrieveSection.propTypes = {
  changeAuthPassword: PropTypes.func,
  checkIsValidUser: PropTypes.func,
  isValidUser: PropTypes.bool,
  requested: PropTypes.bool,
  requestSent: PropTypes.bool,
  errorMessage: PropTypes.string,
  history: PropTypes.shape({
    push: PropTypes.func
  }).isRequired,
  location: PropTypes.shape({
    search: PropTypes.string
  }).isRequired
};
