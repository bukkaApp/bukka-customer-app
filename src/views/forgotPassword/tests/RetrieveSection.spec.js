import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import RetrieveSection from '../components/RetrieveSection';
import auth from './__mocks__/changePasswordAction.json';


const initialState = {
  navbarAuthReducer: { type: '/login' },
  validateTokenReducer: { isValidUser: true, requested: true, },
  changePasswordReducer: { requested: true, errorMessage: 'error occur' },
  authenticationReducer: {
    user: {},
    status: {
      authenticated: false,
      error: false
    },
    errorMessage: ''
  }
};

const middlewares = [];
const mockStore = configureStore(middlewares);
const store = mockStore(initialState);

describe('RetrieveSection component', () => {
  const props = {
    status: {
      authenticated: false,
      error: false
    },
    requested: false,
    requestSent: false,
    errorMessage: 'error occur',
    history: { push: jest.fn(), },
    location: { search: auth.token },
    changeAuthPassword: jest.fn(),
  };
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <RetrieveSection {...props} />
      </MemoryRouter>
    </Provider>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
