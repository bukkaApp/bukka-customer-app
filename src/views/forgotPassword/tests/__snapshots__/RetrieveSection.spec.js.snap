// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrieveSection component renders properly 1`] = `ReactWrapper {}`;
